backend/src/Controller/Admin/CaptainOffer/AdminCaptainOfferController.php

```diff
@@ -0,0 +1,258 @@
+{
+    private AutoMapping $autoMapping;
+    private ValidatorInterface $validator;
+    private AdminCaptainOfferService $adminCaptainOfferService;
+
+    public function __construct(SerializerInterface $serializer, AutoMapping $autoMapping, ValidatorInterface $validator, AdminCaptainOfferService $adminCaptainOfferService)
+    {
+        parent::__construct($serializer);
+        $this->autoMapping = $autoMapping;
+        $this->validator = $validator;
+        $this->adminCaptainOfferService = $adminCaptainOfferService;
+    }
+
+    /**
+     * admin:Create Captain Offer By Admin.
+     * @Route("create", name="createCaptainOfferByAdmin", methods={"POST"})
+     * @IsGranted("ROLE_ADMIN")
+     * @param Request $request
+     * @return JsonResponse
+     *
+     * @OA\Tag(name="Captain Offer")
+     *
+     * @OA\Parameter(
+     *      name="token",
+     *      in="header",
+     *      description="token to be passed as a header",
+     *      required=true
+     * )
+     *
+     * @OA\RequestBody(
+     *      description="new captain offer create request",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="integer", property="carCount"),
+     *          @OA\Property(type="string", property="status"),
+     *          @OA\Property(type="integer", property="expired"),
+     *          @OA\Property(type="number", property="price"),
+     *      )
+     * )
+     *
+     * @OA\Response(
+     *      response=200,
+     *      description="Returns new captain offer create",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code"),
+     *          @OA\Property(type="string", property="msg"),
+     *          @OA\Property(type="object", property="Data",
+     *            @OA\Property(type="integer", property="id"),
+     *            @OA\Property(type="integer", property="carCount"),
+     *            @OA\Property(type="string", property="status"),
+     *           @OA\Property(type="integer", property="expired"),
+     *           @OA\Property(type="number", property="price"),
+     *      )
+     *   )
+     * )
+     *
+     * @Security(name="Bearer")
+     */
+    public function createCaptainOfferByAdmin(Request $request): JsonResponse
+    {
+        $data = json_decode($request->getContent(), true);
+
+        $request = $this->autoMapping->map(stdClass::class, CaptainOfferCreateRequest::class, (object)$data);
+
+        $violations = $this->validator->validate($request);
+
+        if (\count($violations) > 0) {
+            $violationsString = (string) $violations;
+
+            return new JsonResponse($violationsString, Response::HTTP_OK);
+
+        }
+         
+        $result = $this->adminCaptainOfferService->createCaptainOfferByAdmin($request);
+
+        return $this->response($result, self::CREATE);
+    }
+
+    /**
+     * admin:Update Captain Offer By Admin.
+     * @Route("update", name="updateCaptainOfferByAdmin", methods={"PUT"})
```

Comment:
Let us name the API as its aim, like "captainoffer", instead of naming it create or update.